### PR TITLE
Feature/cpl resource enhancement

### PR DESCRIPTION
**Pull request by svenkatrav**

### CODE REVIEW

File: src/main/java/com/netflix/imflibrary/st2067_2/BaseResourceType.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+package com.netflix.imflibrary.st2067_2;
+
+import javax.annotation.concurrent.Immutable;
+import java.math.BigInteger;
+import java.util.List;
+
+/**
+ * A class that models an IMF Base Resource.
+ */
+@Immutable
```
Comment (rpuri-netflix) [inline src/main/java/com/netflix/imflibrary/st2067_2/BaseResourceType.java]: @svenkatrav IMO, a class that is not final should not be annotated as "Immutable"

File: src/main/java/com/netflix/imflibrary/st2067_2/IMFCompositionPlaylistType.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+import javax.annotation.concurrent.Immutable;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * A class that models an IMF Composition Playlist structure.
+ */
+@Immutable
+public final class IMFCompositionPlaylistType {
```
Comment (rpuri-netflix) [inline src/main/java/com/netflix/imflibrary/st2067_2/IMFCompositionPlaylistType.java]: @svenkatrav I am not sure if a class with fields that have "protected" access can be annotated as "Immutable"

**Comment by rpuri-netflix** [comment]
@svenkatrav would it make sense to move "IMFTrackFileResource_st2067_2_2013.java" along side its 2013 CPL counterpart?

File: src/main/java/com/netflix/imflibrary/st2067_2/BaseResourceType.java (hunk 1 above)
Comment (svenkatrav) [inline src/main/java/com/netflix/imflibrary/st2067_2/BaseResourceType.java]: @rpuri-netflix Yes I agree. I have made BaseResourceType as abstract and removed Immutability.

File: src/main/java/com/netflix/imflibrary/st2067_2/IMFCompositionPlaylistType.java (hunk 2 above)
Comment (svenkatrav) [inline src/main/java/com/netflix/imflibrary/st2067_2/IMFCompositionPlaylistType.java]: Yes fixed.

**Pull request closed without merging**